apps/hannk/configure_cmake.sh

```diff
@@ -2,79 +2,106 @@
 
 set -e
 
-HANNK_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+HANNK_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
 
-if [ -z ${BUILD_DIR} ]; then
-BUILD_DIR="${HANNK_DIR}/build"
+if [ -z "${BUILD_DIR}" ]; then
+  BUILD_DIR="${HANNK_DIR}/build"
 fi
 
-if [ -z ${HALIDE_INSTALL_PATH} ]; then
-HALIDE_INSTALL_PATH=${HOME}/halide-14-install/
+if [ -z "${HALIDE_INSTALL_PATH}" ]; then
+  HALIDE_INSTALL_PATH="${HOME}/halide-14-install/"
 fi
 
-if [ -z ${HL_TARGET} ]; then
-HL_TARGET=host
+if [ -z "${HL_TARGET}" ]; then
+  HL_TARGET=host
 fi
 
 if [ -z "${CMAKE_GENERATOR}" ]; then
-CMAKE_GENERATOR=Ninja
+  CMAKE_GENERATOR=Ninja
 fi
 
 if [ -z "${CMAKE_BUILD_TYPE}" ]; then
-CMAKE_BUILD_TYPE=Release
+  CMAKE_BUILD_TYPE=Release
+fi
+
+if [ -z "${ANDROID_PLATFORM}" ]; then
+  ANDROID_PLATFORM=21
 fi
 
 if [ -z "${HANNK_BUILD_TFLITE}" ]; then
-HANNK_BUILD_TFLITE=ON
+  HANNK_BUILD_TFLITE=ON
 else
-HANNK_BUILD_TFLITE=OFF
+  HANNK_BUILD_TFLITE=OFF
 fi
 
-PREFIX=
-EXTRAS=
+## In a cross-compiling scenario, use a separate host and build dir
 
 if [[ "${HL_TARGET}" =~ ^arm-64-android.* ]]; then
-
-  # TODO: this doesn't work (yet); crosscompiling in CMake is painful.
-  echo Configuring for Android arm64-v8a build...
-  echo Using ANDROID_NDK_ROOT=${ANDROID_NDK_ROOT}
-  echo Using CMAKE_TOOLCHAIN_FILE=${ANDROID_NDK_ROOT}/build/cmake/android.toolchain.cmake
-
-  EXTRAS="-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE} -DANDROID_ABI=arm64-v8a"
-
+  HOST_BUILD_DIR="${BUILD_DIR}/_host"
+  HOST_BUILD_TARGET=(--target hannk_tools)
+  HL_HOST_TARGET=host
 elif [[ "${HL_TARGET}" =~ ^wasm-32-wasmrt.* ]]; then
-
-    PREFIX=emcmake
-    HANNK_BUILD_TFLITE=OFF
-
+  HOST_BUILD_DIR="${BUILD_DIR}/_host"
+  HOST_BUILD_TARGET=(--target hannk_tools)
+  HL_HOST_TARGET=host
 else
+  HOST_BUILD_DIR="${BUILD_DIR}"
+  HOST_BUILD_TARGET=()
+  HL_HOST_TARGET="${HL_TARGET}"
+fi
 
-  echo Assuming host build...
+## Build HANNK for the host no matter what
 
-fi
+echo "Configuring HANNK for ${HL_HOST_TARGET}"
+cmake \
+  -G "${CMAKE_GENERATOR}" \
+  -S "${HANNK_DIR}" \
+  -B "${HOST_BUILD_DIR}" \
+  -DCMAKE_BUILD_TYPE="${CMAKE_BUILD_TYPE}" \
+  -DHalide_DIR="${HALIDE_INSTALL_PATH}/lib/cmake/Halide" \
+  -DHalideHelpers_DIR="${HALIDE_INSTALL_PATH}/lib/cmake/HalideHelpers" \
+  -DHalide_TARGET="${HL_HOST_TARGET}" \
+  -DHANNK_BUILD_TFLITE=${HANNK_BUILD_TFLITE}
 
-if [ -n "${NODE_JS_EXECUTABLE}" ]; then
-EXTRAS="${EXTRAS} -DNODE_JS_EXECUTABLE=${NODE_JS_EXECUTABLE}"
-echo Using NODE_JS_EXECUTABLE=${NODE_JS_EXECUTABLE}
-fi
+echo "Building HANNK for ${HL_HOST_TARGET}"
```

Comment:
> Also, a nit: do we actually need to do this explicit build if we are crosscompiling? Shouldn't the later invocation of cmake --build also trigger what we need? Or is it just needed to prime the pump the first time, for reasons that aren't clear to me? (Note that it would be undesirable to require re-running this script to properly rebuild the world. I would expect to be able to do ninja -t clean && ninja all to safely rebuild the world for both host and crosscompiles.)

As it's designed right now, we do need a re-build. I can work on a v2 that doesn't require this (even before landing this, if you want).

> (Echoing the other build settings (eg HL_TARGET) is (IMHO) a desirable thing since they may get inferred from pre-set env vars -- was removing those deliberate or just a byproduct?)

Just a byproduct, sorry.